### ISSUE TITLE
layers control closes too quickly

### ISSUE THREAD

**Issue by fyyyyy** [Leaflet/Leaflet#9611]
### Checklist

- [x] I've looked at the [documentation](https://leafletjs.com/reference.html) to make sure the behavior isn't documented and expected.
- [x] I'm sure this is an issue with Leaflet, not with my app or other dependencies (Angular, Cordova, React, etc.).
- [x] I've searched through the [current issues](https://github.com/Leaflet/Leaflet/issues) to make sure this hasn't been reported yet.
- [x] I agree to follow the [Code of Conduct](https://github.com/Leaflet/Leaflet/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.

### Steps to reproduce

Users are complaining that the layer control disappears too quickly. Especially when you have a long list of layers, its quite tedious to move the mouse cursor down without leaving the hover state. A transition or transition-delay would help, however I have been unable to implement this with pure css. 

I have found a workaround by triggering the collapse manually, however its causes other issues, like re-entering while not clearing older timeouts. Probably needs a debounce fn.
https://plnkr.co/edit/WQyfFctUqs7mNdq9

On another note, a collapseDelay option would work really well to make it customizable.

### Expected behavior

Layer control stays for 1-2 seconds after leaving hover state.

### Current behavior

Layer control disappears instantly after leaving hover state

### Environment

- Leaflet version: 1.9.3
- Browser (with version): Chrome
- OS/Platform (with version): Mac/Windows latest

**Comment by fyyyyy** [comment]
> [@fyyyyy](https://github.com/fyyyyy) This is kinda easy, so I would encourage you to try and send a PR for it.

The PR and test is ready, just needs review
